cosmos-tx/tests/integration.rs

```diff
@@ -0,0 +1,191 @@
+        .account_id(ACCOUNT_PREFIX)
+        .unwrap();
+
+    let amount = Coin {
+        amount: 1u8.into(),
+        denom: DENOM.parse().unwrap(),
+    };
+
+    let msg_send = MsgSend {
+        from_address: sender_account_id.clone(),
+        to_address: recipient_account_id,
+        amount: vec![amount.clone()],
+    }
+    .to_msg()
+    .unwrap();
+
+    let chain_id = CHAIN_ID.parse().unwrap();
+    let sequence_number = 0;
+    let gas = 100_000;
+    let fee = Fee::from_amount_and_gas(amount, gas);
+    let timeout_height = 9001u16;
+    let tx_body = tx::Body::new(vec![msg_send], MEMO, timeout_height);
+
+    let tx = Builder::new(chain_id, ACCOUNT_NUMBER)
+        .sign_tx(tx_body, &sender_private_key, sequence_number, fee)
+        .unwrap();
+
+    let docker_args = [
+        "-d",
+        "-p",
+        &format!("{}:{}", RPC_PORT, RPC_PORT),
+        DOCKER_IMAGE,
+        CHAIN_ID,
+        &sender_account_id.to_string(),
+    ];
+
+    docker_run(&docker_args, || {
+        init_tokio_runtime().block_on(async {
+            let rpc_address = format!("http://localhost:{}", RPC_PORT);
+            let rpc_client = rpc::HttpClient::new(rpc_address.as_str()).unwrap();
+
+            rpc_client
+                .wait_until_healthy(Duration::from_secs(5))
+                .await
+                .unwrap();
+
+            // Workaround for what appears to be a Tendermint race condition.
+            // There is a short period after the RPC health check succeeds
+            // and the `/genesis` endpoint returns valid data where
+            // transactions fail due to an internal SDK panic in the
+            // "undefined" code space:
+            //
+            //     TxResult {
+            //         code: Err(111222), data: None, log: Log("panic"), info: Info(""),
+            //         gas_wanted: Gas(0), gas_used: Gas(0), events: [],
+            //         codespace: Codespace("undefined")
+            //     }
+            //
+            // This should ideally get fixed upstream, however there is
+            // presently no tracking issue for this bug.
+            //
+            // If we see it pop up locally or in CI, perhaps we could find
+            // some way of polling readiness via RPC, or retrying when this
+            // specific error is encountered.
+            // TODO(tarcieri): open upstream issue and remove this hack when fixed
+            tokio::time::sleep(Duration::from_secs(2)).await;
+
+            let tx_commit_response = tx.broadcast_commit(&rpc_client).await.unwrap();
+
+            if tx_commit_response.check_tx.code.is_err() {
+                panic!("check_tx failed: {:?}", tx_commit_response.check_tx);
+            }
+
+            if tx_commit_response.deliver_tx.code.is_err() {
+                panic!("deliver_tx failed: {:?}", tx_commit_response.deliver_tx);
+            }
+
+            // TODO(tarcieri): look up transaction by hash and test transaction parsing
```

Comment:
I have the lookup grpc code, should I just upstream that as a broken out cate? I can integrate it in here.

Reply:
Since the tests are already using RPC, I can just hit the `/block` endpoint:

https://docs.rs/tendermint-rpc/0.19.0/tendermint_rpc/trait.Client.html#method.latest_block

Reply:
Since we already have an RPC client instantiated, I can use the `/tx_search` endpoint to look it up by hash:

https://docs.rs/tendermint-rpc/0.19.0/tendermint_rpc/trait.Client.html#method.tx_search